### PR TITLE
Send application performance metrics to Datadog

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.shortcut.com/simpledotorg/story/5505/enable-end-to-end-application-monitoring

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -455,6 +456,8 @@ dependencies {
   lintChecks(projects.lint)
 
   runtimeOnly(libs.jackson.core)
+
+  implementation(libs.datadog.sdk)
```
Comment (msasikanth) [inline app/build.gradle.kts:464]: Just a personal preference, we can probably add dependencies in alphabetical order. 😅

File: app/src/main/java/org/simple/clinic/ClinicApp.kt [hunk 2]
```diff
@@ -86,6 +91,11 @@ abstract class ClinicApp : Application(), CameraXConfig.Provider {
   }
 
   private fun setupApplicationPerformanceMonitoring() {
+    val samplingRate = remoteConfig
+        .double("datadog_sample_rate", 0.0)
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/ClinicApp.kt:95]: Currently we have this turned off?

File: app/build.gradle.kts (hunk 1 above)
Comment (vinaysshenoy) [reply app/build.gradle.kts:464]: Hmm, I prefer to add new dependencies at the bottom of the file, so it matches the version control history 😅

File: app/src/main/java/org/simple/clinic/ClinicApp.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/ClinicApp.kt:95]: It's turned off by default, I have enabled it on Remote Config consoles.

**Review by msasikanth — APPROVED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (msasikanth) [reply app/build.gradle.kts:464]: Yeah, I figured to keep this alphabetical to make it easier to find dependencies. But sure, this works. We can refactor it once in a while I guess.

**Review by msasikanth — APPROVED** [review]

**Pull request merged**